docs/core/transactions/confirmation.md

```diff
@@ -59,61 +70,65 @@ hashes to build a trusted clock. The “history” part of the name comes from t
-
-Transaction not expired pseudocode:
-`currentBankSlot - slotForTxRecentBlockhash < 152`
+timestamp and expires when that blockhash is no longer “recent enough”.
+
+As each block is finalized (i.e. the maximum tick height
+[is reached](https://github.com/anza-xyz/agave/blob/0588ecc6121ba026c65600d117066dbdfaf63444/runtime/src/bank.rs#L3269-L3271),
+reaching the "block boundary"), the final hash of the block is added to the
+`BlockhashQueue` which stores a maximum of the
+[300 most recent blockhashes](https://github.com/anza-xyz/agave/blob/e0b0bcc80380da34bb63364cc393801af1e1057f/sdk/program/src/clock.rs#L123-L126).
+During transaction processing, Solana Validators will check if each
+transaction's recent blockhash is recorded within the
+[150 most recent blockhashes](https://github.com/anza-xyz/agave/blob/cb2fd2b632f16a43eff0c27af7458e4e97512e31/sdk/program/src/clock.rs#L129-L131)
+(aka the max processing age) stored within the `BlockhashQueue`. If the
+transaction's recent blockhash is
+[older than this](https://github.com/anza-xyz/agave/blob/cb2fd2b632f16a43eff0c27af7458e4e97512e31/runtime/src/bank.rs#L3570-L3571)
+max processing age, the transaction is not processed.
+
+Since [slots](/docs/terminology.md#slot) (aka the time period a validator can
+produce a block) are configured to last about
+[400ms](https://github.com/anza-xyz/agave/blob/cb2fd2b632f16a43eff0c27af7458e4e97512e31/sdk/program/src/clock.rs#L107-L109),
+but may fluctuate between 400ms and 600ms, a given blockhash can only be used by
+transactions for about 60 to 90 seconds before it will be considered expired by
+the runtime.
 
 ### Example of transaction expiration
 
 Let’s walk through a quick example:
 
-1. A validator is producing a new block for slot #1000
-2. The validator receives a transaction with recent blockhash `1234...` from a
-   user
-3. The validator checks the `1234...` blockhash against the list of recent
-   blockhashes leading up to its new block and discovers that it was the
-   blockhash for slot #849
-4. Since slot #849 is exactly 151 slots lower than slot #1000, the transaction
-   hasn’t expired yet and can still be processed!
-5. But wait, before actually processing the transaction, the validator finished
-   the block for slot #1000 and starts producing the block for slot #1001
-   (validators get to produce blocks for 4 consecutive slots).
-6. The validator checks the same transaction again and finds that it’s now too
-   old and drops it because it’s now 152 slots lower than the current slot :(
+1. A validator is actively producing a new block for the current slot
+2. The validator receives a transaction from a user with the recent blockhash
+   `abcd...`
+3. The validator checks this blockhash `abcd...` against the list of recent
+   blockhashes in the `BlockhashQueue` and discovers that it was created 150
+   blocks ago
+4. Since it was created exactly 150 blocks ago, the transaction has not expired
+   yet and can still be processed!
```

Comment:
Up to you if you want to include the off-by-one-ness and say 151

Reply:
I will also clarify this in a new PR! Making it more explicitly clear